packages/discord.js/src/managers/GuildBanManager.js

```diff
@@ -199,6 +199,51 @@ class GuildBanManager extends CachedManager {
     await this.client.rest.delete(Routes.guildBan(this.guild.id, id), { reason });
     return this.client.users.resolve(user);
   }
+
+  /**
+   * Options used for bulk banning users from a guild.
+   * @typedef {Object} BanBulkOptions
+   * @property {number} [deleteMessageSeconds] Number of seconds of messages to delete,
+   * must be between 0 and 604800 (7 days), inclusive
+   * @property {string} [reason] The reason for the ban
+   */
+
+  /**
+   * Result of bulk banning users from a guild.
+   * @typedef {Object} BanBulkResult
+   * @property {Snowflake[]} bannedUsers IDs of the banned users
+   * @property {Snowflake[]} failedUsers IDs of the users that could not be banned or were already banned
+   */
+
+  /**
+   * Bulk ban users from a guild, and optionally delete previous messages sent by them.
+   * @param {Collection<Snowflake, UserResolvable>|UserResolvable[]} users The users to ban
+   * @param {BanBulkOptions} [options] The options for bulk banning users
+   * @returns {Promise<BanBulkResult>} Returns an object with `bannedUsers` key containing the IDs of the banned users
+   * and the key `failedUsers` with the IDs that could not be banned or were already banned.
+   * @example
+   * // Bulk ban users by ids (or with user/guild member objects)
```

Comment:
```suggestion
   * // Bulk ban users by ids (or with user/guild member objects) and delete all their messages from the past 7 days
```

probably you can come up with something better